lib/base/configuration.cpp

```diff
@@ -12,7 +30,95 @@ String Configuration::ApiBindHost;
 String Configuration::ApiBindPort{"5665"};
 bool Configuration::AttachDebugger{false};
 String Configuration::CacheDir;
-int Configuration::Concurrency{static_cast<int>(std::thread::hardware_concurrency())};
+
+#ifdef __linux__
+static std::string ReadSysLine(const char *file)
+{
+	using namespace boost::asio;
+	using namespace boost::system;
+
+	int f = open(file, O_RDONLY);
+
+	if (f < 0) {
+		if (errno == ENOENT) {
+			return "";
+		}
+
+		throw system_error(error_code(errno, system_category()));
```

Comment:
Do we really want to fail starting in this case? I think logging a warning and continuing with the value from `std::thread::hardware_concurrency()` would be a better option.

Reply:
1. In "this case" a kernel subsystem(!) is out of service. The best reason to panic.
2. Logging? We're in a static constructor.